sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroSource.java

```diff
@@ -270,20 +276,19 @@ public void validate() {
       }
       codec = metadata.getCodec();
       syncMarker = metadata.getSyncMarker();
-      fileSchemaString = metadata.getSchemaString();
-      // If the source was created with a null schema, use the schema that we read from the file's
-      // metadata.
-      if (readSchemaString == null) {
-        readSchemaString = metadata.getSchemaString();
+
+      // Only store both the read and file schema if they differ
```

Comment:
Is this to save memory?
Does it matter if readSchemaString references to the same object.
I think store a reference will make other code easy to read.